src/Services/PaymentReferenceCodeGenerator/RandomPaymentReferenceCodeGenerator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\PaymentContext\Services\PaymentReferenceCodeGenerator;
+
+use WMDE\Fundraising\PaymentContext\Domain\CharacterIndexGenerator;
+use WMDE\Fundraising\PaymentContext\Domain\ChecksumGenerator;
+use WMDE\Fundraising\PaymentContext\Domain\Model\PaymentReferenceCode;
+use WMDE\Fundraising\PaymentContext\Domain\PaymentReferenceCodeGenerator;
+
+class RandomPaymentReferenceCodeGenerator implements PaymentReferenceCodeGenerator {
+
+	private CharacterIndexGenerator $characterIndexGenerator;
+
+	/**
+	 * @var string[]
+	 */
+	protected array $characters;
+	protected int $characterCount;
+	private int $maxRandom;
+
+	public function __construct( CharacterIndexGenerator $characterIndexGenerator ) {
+		$this->characterIndexGenerator = $characterIndexGenerator;
+		$this->characters = str_split( PaymentReferenceCode::ALLOWED_CHARACTERS );
+		$this->characterCount = count( $this->characters );
```

Comment:
This is unused

Reply:
weird that PhpStorm didn't point that out